app/app_main.c

```diff
@@ -2712,61 +2744,81 @@ static ACVP_RESULT app_rsa_sigver_handler(const unsigned char* msg, const int ms
+        /*rsa_pkctx = EVP_PKEY_CTX_new(rsa_evpkey, NULL);
         if (!rsa_pkctx) {
-            printf( "ERROR: EVP_PKEY_CTX_new() failed\n");
+            printf( "ERROR: EVP_PKEY_CTX_new() failed in RSA SigVer test case.\n");
             goto err;
-        }
+        }*/
 
         long lerr;
-        /* Initialize operation  */
+        // Initialize operation
         ret = EVP_DigestVerifyInit(tc_mdctx, &rsa_pkctx, tc_md, NULL, rsa_evpkey);
         if (ret != 1) {
-            printf("ERROR: EVP_DigestVerifyInit failed.\n");
+            printf("ERROR: EVP_DigestVerifyInit failed in RSA SigVer test case.\n");
             while ((lerr = ERR_get_error()))
                 printf( "ERROR:%s\n", ERR_error_string(lerr, NULL));
             goto err;
         }
 
-        /* set padding */
+        // set padding
         ret = EVP_PKEY_CTX_set_rsa_padding(rsa_pkctx, padding);
         if (ret != 1) {
-            printf("ERROR: EVP_PKEY_CTX_set_rsa_padding failed.\n");
+            printf("ERROR: EVP_PKEY_CTX_set_rsa_padding failed in RSA SigVer test case.\n");
             while ((lerr = ERR_get_error()))
                 printf( "ERROR:%s\n", ERR_error_string(lerr, NULL));
             goto err;
         }
 
-        /* update the message */
-        ret = EVP_DigestVerifyUpdate(tc_mdctx, msg, msg_len);
+        /*ret = EVP_PKEY_CTX_set_rsa_pss_saltlen(rsa_pkctx, 0);
+        if (ret != 1) {
+            printf("ERROR: EVP_PKEY_CTX_set_rsa_pss_saltlen failed in RSA SigVer test case.\n");
+            while ((lerr = ERR_get_error()))
+                printf( "ERROR:%s\n", ERR_error_string(lerr, NULL));
+            goto err;
+        }*/
+        // update the message
+        ret = EVP_DigestVerifyUpdate(tc_mdctx, msg, msglen);
         if (ret != 1) {
-            printf("ERROR: EVP_DigestVerifyUpdate failed.\n");
+            printf("ERROR: EVP_DigestVerifyUpdate failed in RSA SigVer test case.\n");
             while ((lerr = ERR_get_error()))
                 printf( "ERROR:%s\n", ERR_error_string(lerr, NULL));
             goto err;
         }
 
-        /* Finalize */
-        ret = EVP_DigestVerifyFinal(tc_mdctx, sig, sig_len);
-        if (ret != 1) {
-            printf("ERROR: RSA Digest verify failed.\n");
+        // Finalize
+        ret = EVP_DigestVerifyFinal(tc_mdctx, sig, siglen);
+        switch(ret) {
+        case 0:
+            ret = 1;
+            break;
+        case 1:
+            sig_tc->pass = 1;
+            break;
+        default:
+            printf("ERROR: EVP_DigestVerifyFinal failed in RSA SigVer test case.\n");
             while ((lerr = ERR_get_error()))
                 printf( "ERROR:%s\n", ERR_error_string(lerr, NULL));
             goto err;
         }
+
+    }
+    else {
+        ret = 0;
     }
 
 err:
-    if (rsa_key) RSA_free(rsa_key);
+    if (msg) free(msg);
+    if (sig) free(sig);
     if (rsa_evpkey) EVP_PKEY_free(rsa_evpkey);
+    if (rsa_pkctx) {
```

Comment:
I think these lines should be deleted as well. That struct should be cleaned up as part of tc_mdctx.